libcaf_core/tests/legacy/dsl.cpp

```diff
@@ -252,7 +252,7 @@ SCENARIO("passing requirements increment the 'good' counter") {
       auto silent_expr = [&] {                                                 \
         auto bk = caf::test::logger::instance().make_quiet();                  \
         auto guard = caf::detail::make_scope_guard(                            \
-          [bk] { caf::test::logger::instance().levels(bk); });                 \
+          [bk]() noexcept { caf::test::logger::instance().levels(bk); });      \
```

Comment:
Adding `noexcept` here requires that we also add `noexcept` to `levels` in `libcaf_test/caf/test/unit_test.hpp`.